src/onefetch/language.rs

```diff
@@ -163,18 +163,36 @@ impl Language {
     fn get_languages_stat(languages: &tokei::Languages) -> Option<HashMap<Language, f64>> {
         let mut stats = HashMap::new();
 
-        let sum_language_code: usize = languages.iter().map(|(_, v)| v.code).sum();
+        for (language_type, language) in languages.iter() {
+            let mut code = language.code;
 
-        if sum_language_code == 0 {
+            let has_children = !language.children.is_empty();
+
+            if has_children {
+                for reports in language.children.values() {
+                    for stats in reports.iter().map(|r| r.stats.summarise()) {
+                        code += stats.code;
+                    }
+                }
+            }
+
+            if code == 0 {
+                continue;
+            }
+
+            stats.insert(Language::from(*language_type), code as f64);
+        }
+
+        let total: f64 = stats.iter().map(|(_, v)| v).sum();
+
+        if (total - 0 as f64).abs() < f64::EPSILON {
```

Comment:
Is it necessary to subtract by 0? :thinking:

Reply:
It looks weird, I aggree; it's a clippy suggestion.

At first, I had written `total == 0 as f64`

Reply:
Got it, that makes sense. I assume clippy wouldn't complain if it was this
```suggestion
        if total.abs() < f64::EPSILON {
```

Reply:
I should really stop relying blindly on clippy suggestions :weary:
Thanks @spenserblack